Postinstall script installation fails when installing locally

<!--
1. If you have a question and not a bug/feature request please ask it at http://forum.serverless.com
2. Please check if an issue already exists so there are no duplicates
3. Check out and follow our Guidelines: https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md
4. Fill out the whole template so we have a good overview on the issue
5. Do not remove any section of the template. If something is not applicable leave it empty but leave it in the Issue
6. Please follow the template, otherwise we'll have to ask you to update it
-->

# This is a Bug Report

## Description
I know a correct way to install globally using `npm install serverless - g`, but when installing locally in own project with `npm install serverless --save`, an error occurs during execution of postinstall.js.

Is this bug? or does it mean sls has not supported for installing locally?

### step to reproduce
- Create directory with `mkdir sls-project && cd sls-project`
- Run `npm init`
- Run `npm install serverless --save`, and you will see the following error

```
$npm install serverless --save
npm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning EINTEGRITY: sha1-F2n0tVHu3OQZ8FBd6uLiZ2NULTc= integrity checksum failed when using sha1: wanted sha1-F2n0tVHu3OQZ8FBd6uLiZ2NULTc= but got sha512-0TZ20a+xcIl4u0+Mj5xDH2yOWdmQiXlKf9Hm+TgDXjTMsEYb+gDrmb8e8UNAzMCitX8NBqG4Z/FUQIyzv/R1JQ==. (2982 bytes)

> serverless@1.15.3 postinstall /Users/horike/src/sls-project/node_modules/serverless
> node ./scripts/postinstall.js

module.js:471
    throw err;
    ^

Error: Cannot find module '/Users/horike/src/sls-project/node_modules/serverless/node_modules/tabtab/src/cli.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
module.js:471
    throw err;
    ^

Error: Cannot find module '/Users/horike/src/sls-project/node_modules/serverless/node_modules/tabtab/src/cli.js'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3

> spawn-sync@1.0.15 postinstall /Users/horike/src/sls-project/node_modules/spawn-sync
> node postinstall

npm WARN saveError ENOENT: no such file or directory, open '/Users/horike/src/sls-project/package.json'
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN enoent ENOENT: no such file or directory, open '/Users/horike/src/sls-project/package.json'
npm WARN sls-project No description
npm WARN sls-project No repository field.
npm WARN sls-project No README data
npm WARN sls-project No license field.

+ serverless@1.15.3
added 210 packages in 23.298s
```
Similar or dependent issues:
* #3760
* #3763

## Additional Data
* ***node version***: 6.10.2
* ***npm version***: 5.0.3
* ***Serverless Framework Version you're using***:1.15.3
* ***Operating System***: macOS 10.11.6
* ***Stack Trace***: above
* ***Provider Error messages***: above

--- a/scripts/postinstall.js
+++ b/scripts/postinstall.js
@@ -5,7 +5,6 @@
 
 const Serverless = require('../lib/Serverless');
 const execSync = require('child_process').execSync;
-const path = require('path');
 
 try {
   const serverless = new Serverless();

--- a/scripts/postinstall.js
+++ b/scripts/postinstall.js
@@ -5,6 +5,8 @@
 
 const Serverless = require('../lib/Serverless');
 const execSync = require('child_process').execSync;
+const path = require('path');
+const fileExistsSync = require('../lib/utils/fs/fileExistsSync');
 
 try {
   const serverless = new Serverless();
@@ -20,13 +22,17 @@ try {
 
 function setupAutocomplete() {
   return new Promise((resolve, reject) => {
+    let tabtabPath = './node_modules/tabtab';
+    if (!fileExistsSync(path.join(__dirname, '..', 'node_modules', 'tabtab', 'src', 'cli.js'))) {
+      tabtabPath = '../tabtab';
+    }
     try {
-      execSync('node ./node_modules/tabtab/src/cli.js install --name serverless --auto');
-      execSync('node ./node_modules/tabtab/src/cli.js install --name sls --auto');
+      execSync(`node ${tabtabPath}/src/cli.js install --name serverless --auto`);
+      execSync(`node ${tabtabPath}/src/cli.js install --name sls --auto`);
       return resolve();
     } catch (error) {
-      execSync('node ./node_modules/tabtab/src/cli.js install --name serverless --stdout');
-      execSync('node ./node_modules/tabtab/src/cli.js install --name sls --stdout');
+      execSync(`node ${tabtabPath}/src/cli.js install --name serverless --stdout`);
+      execSync(`node ${tabtabPath}/src/cli.js install --name sls --stdout`);
       console.log('Could not auto-install serverless autocomplete script.');
       console.log('Please copy / paste the script above into your shell.');
       return reject(error);

--- a/scripts/postinstall.js
+++ b/scripts/postinstall.js
@@ -6,7 +6,6 @@
 const Serverless = require('../lib/Serverless');
 const execSync = require('child_process').execSync;
 const path = require('path');
-const fileExistsSync = require('../lib/utils/fs/fileExistsSync');
 
 try {
   const serverless = new Serverless();
@@ -22,10 +21,7 @@ try {
 
 function setupAutocomplete() {
   return new Promise((resolve, reject) => {
-    let tabtabPath = './node_modules/tabtab';
-    if (!fileExistsSync(path.join(__dirname, '..', 'node_modules', 'tabtab', 'src', 'cli.js'))) {
-      tabtabPath = '../tabtab';
-    }
+    const tabtabPath = require.resolve('tabtab').replace(/\/index.js$/, '');
     try {
       execSync(`node ${tabtabPath}/src/cli.js install --name serverless --auto`);
       execSync(`node ${tabtabPath}/src/cli.js install --name sls --auto`);

